ipaserver/plugins/cert.py

```diff
@@ -860,6 +866,33 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         )
 
 
+def _emails_are_valid(subject_emails, san_emails, principal_emails):
+    """
+    Checks if any email addr from DN or SAN extension does not
+    appear in ldap entry, comparing the domain part case-insensitively.
+    """
+
+    if not any(principal_emails):
+        return False
+
+    def lower_domain(email):
+        if '@' not in email:
+            return ''
+
+        email_splited = email.split('@')
+        return '{}@{}'.format(email_splited[0], email_splited[1].lower())
```

Comment:
This logic still needs to be fixed.  It is simple to construct a CSR with emails that have no `@` (or multiple).
(It probably possible to make a bogus LDAP attribute value, too).

So I'd suggest something like:

```python
email_split = email.split('@', 1)  # split only at first occurrence of '@'
if len(email_split) > 1:
    email_split[1] = email_split[1].lower()
return email_split
```

There is no need to "rejoin" the parts back into a string; we can compare on the resulting
lists.  If you end up using `set` then you can turn the lists into tuples so that you have a
hashable value.